### PR TITLE
Add extract API to Go SDK

### PR DESCRIPTION
**Pull request by mkozakov**
This PR adds the Extract V0 API to the Go SDK

### CODE REVIEW

File: client_test.go [hunk 1]
```diff
@@ -1,11 +1,10 @@
 package cohere
 
 import (
-	"os"
 	"testing"
 )
 
-var apiKey = os.Getenv("API_KEY")
+var apiKey = "qjGVNgxWsj6pHdXsprfa55Au7msQxS7Igl0Wey2n"
```
Comment (robertkozin) [inline client_test.go:7]: 😅

File: client.go [hunk 2]
```diff
@@ -184,6 +185,23 @@ func (c *Client) Embed(model string, opts EmbedOptions) (*EmbedResponse, error)
 	return ret, nil
 }
 
+// Extracts entities of specified types from the provided text. Each extraction
+// contains a type and a confidence score.
+// See: https://docs.cohere.ai/extract-reference
+// Returns an EmbedResponse object.
```
Comment (robertkozin) [inline client.go:191]: Is this true?

Comment (mkozakov) [reply client.go:191]: - Re: confidence score. Not true yet, but that's what I'm currently working on. 
- EmbedResponse object, not true will fix

File: client_test.go (hunk 1 above)
Comment (mkozakov) [reply client_test.go:7]: Classic. I'm going to force push and also invalidate the API key.

**Review by robertkozin — APPROVED** [review]

**Pull request merged**